scripts/ASTImportTest.sh

```diff
@@ -131,6 +146,125 @@ function test_ast_import_export_equivalence
     TESTED=$((TESTED + 1))
 }
 
+function test_evmjson_import_export_equivalence
+{
+    local sol_file="$1"
+    local input_files=( "${@:2}" )
+    local outputs=( "asm" "bin" "bin-runtime" "opcodes" "srcmap" "srcmap-runtime" )
+    local export_command=("$SOLC" --combined-json "$(IFS=, ; echo "${outputs[*]}")" --pretty-json --json-indent 4 "${input_files[@]}")
+    local success=1
+    if ! "${export_command[@]}" > expected.json 2> expected.error
+    then
+        success=0
+        printError "ERROR: (export) EVM Assembly JSON reimport failed for ${sol_file}"
+        if [[ $EXIT_ON_ERROR == 1 ]]
+        then
+            print_used_commands "$(pwd)" "${export_command[*]}" ""
+            return 1
+        fi
+    fi
+    if ! "${export_command[@]}" "--optimize" > expected.optimize.json 2> expected.optimize.error
+    then
+        success=0
+        printError "ERROR: (export with --optimize) EVM Assembly JSON reimport failed for ${sol_file}"
+        if [[ $EXIT_ON_ERROR == 1 ]]
+        then
+            print_used_commands "$(pwd)" "${export_command[*]} --optimize" ""
+            return 1
+        fi
+    fi
+
+    for input_json in "expected.json" "expected.optimize.json"
+    do
+        local optimize_param=""
+        if [[ "$input_json" == "expected.optimize.json" ]]
+        then
+            optimize_param="--optimize"
+        fi
+
+        # Note that we have some test files, that only consists of free functions.
+        # Those files doesn't define any contracts, so the resulting json does not define any
+        # keys. In this case `jq` returns an error like `jq: error: null (null) has no keys`
+        # to not get spammed by these errors, errors are redirected to /dev/null.
+        for contract in $(jq '.contracts | keys | .[]' $input_json 2> /dev/null)
+        do
+            for output in "${outputs[@]}"
+            do
+                jq --raw-output ".contracts.${contract}.\"${output}\"" "$input_json" > "expected.${output}.json"
+            done
```

Comment:
Instead of processing every output of every contract individually, maybe it would be simpler to just extract those outputs for all contracts into one JSON and compare that?

Here's how you'd do that for this file structure:
```bash
local output_list
output_list=$(IFS=, ; echo "${outputs[*]}")
jq --sort-keys "[.contracts | .[] | {$output_list}]"
```

This should give you an array with one dict per contract and each dict should contain keys for all the outputs from the list (with `null` if the output was not present in the input file).